sdk/python/kfp/v2/compiler/compiler.py

```diff
@@ -684,23 +689,85 @@ def _group_to_dag_spec(
         raise NotImplementedError(
             'dsl.ExitHandler is not yet supported in KFP v2 compiler.')
 
+      importer_tasks = []
```

Comment:
What if we made importer something that requires explicit user action instead? Will that simplify or obviate some of the code below?

Reply:
+1
Let's start with explicit and see the usage patterns.
I'm not sure we can fully support automatic importing now, since the user-provided URIs are not necessarily compatible with the components that work on the URIs and the execution backend lacks resources to detect or work around this incompatibility.

Reply:
Yes, I believe explicit importer behavior will definitely obviate some of the code here. Will follow up offline. (this chunk of code isn't added by this PR, but I'm moving the code up, so that importer tasks are counted in `tasks_in_current_dag`).